lib/dor/datastreams/content_metadata_ds.rb

```diff
@@ -35,10 +35,40 @@ class ContentMetadataDS < ActiveFedora::OmDatastream
 
     def public_xml
       result = ng_xml.clone
-      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")])]'   ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'      ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver').each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file/checksum').each { |n| n.remove }
+      
+      # remove any resources or attributes that are not destined for the public XML
+      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")]|externalFile)]').each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'                ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver'          ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file/checksum'                                              ).each { |n| n.remove }
+      
+      # support for dereferencing links via externalFile element(s) to the source (child) item - see JUMBO-19
+      result.xpath('/contentMetadata/resource/externalFile').each do |externalFile|
+        # enforce pre-conditions that resourceId, objectId, fileId are required
+        src_resource_id = externalFile['resourceId']
+        src_druid = externalFile['objectId']
+        src_file_id = externalFile['fileId']
+        fail ArgumentError, "Malformed externalFile data: #{externalFile.inspect}" if [src_resource_id, src_file_id, src_druid].map(&:blank?).any?
+
+        # grab source item
+        src_item = Dor::Item.find(src_druid)
+        src_item_id = DruidTools::Druid.new(src_item.pid).id # strip druid: prefix
```

Comment:
I'm still concerned about using DruidTools in dor-services.  Given recent discussions about this in slack (which drh has missed while away), this assumption here could be wrong.  That is, @atz noted that dor-services must be allowed to work with things that are not a valid DRUID.  If so, then using it here will break that rule.  I wish it were easier to apply a global rule that everything dor-services handles must be a valid DRUID, so that DruidTools could used consistently (and I'd rather see `src_item.id` method do what we see here).  However....